data_safe_haven/infrastructure/project_manager.py

```diff
@@ -346,6 +346,15 @@ def refresh(self) -> None:
             msg = f"Pulumi refresh failed.\n{exc}"
             raise DataSafeHavenPulumiError(msg) from exc
 
+    def run_pulumi_command(self, command: str) -> str:
+        """Run a Pulumi CLI command using this project and stack."""
+        try:
+            result = self.stack._run_pulumi_cmd_sync(command.split())
```

Comment:
Note that interactive commands won't work as we aren't passing stdin to the Pulumi process.

Reply:
:+1: I've added to the docstring.